tools/data_source/upload.py

```diff
@@ -241,9 +249,17 @@ def add_composite_file(dataset, output_path, files_path):
                         sniff.convert_newlines_sep2tabs(dp, tmp_dir=tmpdir, tmp_prefix=tmp_prefix)
                     else:
                         sniff.convert_newlines(dp, tmp_dir=tmpdir, tmp_prefix=tmp_prefix)
-                shutil.move(dp, os.path.join(files_path, name))
+
+                # move the file to its final destination
+                file_output_path = os.path.join(files_path, name)
+                shutil.move(dp, file_output_path)
+                # groom the dataset file content if required by the corresponding datatype definition
+                if datatype.dataset_content_needs_grooming(file_output_path):
+                    datatype.groom_dataset_content(file_output_path)
```

Comment:
I don't know if this is relevant but if you want to set the name dynamically from https://github.com/workflow4metabolomics/mtbls-dwnld/blob/develop/mtbls-dwnld.xml - I think I could help with that. Rather than change the data type I'd recommend setting the name in a tool provided metadata file - I could give you some links for that. If the problem is really about upload though that isn't helpful.

Reply:
I'm interested in writing the test. I'll have a look at your pointers.

The name of the dataset is set correctly within mtbls-dwnld. So, no, the issue is with the uploader.

Reply:
Hi @jmchilton , I've added a small zip file for testing the uploader, and started to write a method `test_composite_datatype_isatab`. Howerver I haven't figured out yet how to submit the content of the zip file to the uploader (or the file path if this is sufficient). Could you help me please?

Reply:
Hi @jmchilton , I see that our PR is still scheduled for the 18.09 release, however as stated by @sneumann it is really important for us that it gets included in the 18.05 release. The only issue remaining seems to be this uploader test. If you help me on this, I can finish it.

Reply:
@pkrog @jmchilton I've updated the test.